crates/flox-rust-sdk/src/models/flox_package.rs

```diff
@@ -211,9 +211,44 @@ pub enum ParseError {
     NoPackage,
 }
 
+impl Display for FloxTriple {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.stability != Stability::Stable {
+            write!(f, "{}.", self.stability)?;
+        }
+        if self.channel != "nixpkgs-flox" {
+            write!(f, "{}.", self.channel)?;
+        }
+        write!(f, "{}", self.name)?;
+        if let Some(version) = &self.version {
+            write!(f, "@{}", version)?;
+        }
+        Ok(())
+    }
+}
+
 impl Display for FloxPackage {
-    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        todo!()
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FloxPackage::Triple(triple) => write!(f, "{}", triple),
+            _ => todo!(),
+        }
+    }
+}
+
+pub fn packages_to_string(packages: &Vec<FloxPackage>) -> String {
```

Comment:
You don't actually need to take a `&Vec<FloxPackage>` here. Anywhere that you'd use `&Vec<T>` you almost always want `&[T]` instead. It expresses the same intent (an ordered collection of `T`) without needing to be backed by a `Vec` specifically (many types can be coerced or converted to `&[T]` without being backed by `Vec`).

Reply:
Updated to use `&[FloxPackage]`